### ISSUE TITLE
Menu bar focus on Mac

### ISSUE THREAD

**Issue by sofroniewn** [napari/napari#380]
## 🐛 Bug
We've now added a menubar, but you need to toggle focus in and out of napari before it becomes active on the mac. This bug has been encountered in other Qt apps, but we still need to look into fixing.

See here - https://github.com/robotology/yarp/issues/457

**Comment by tlambert03** [comment]
a little history:
- This bug was originally fixed in #553 by removing the call to `self._qt_window.raise_()` in `Window.show`
- Removing `raise_()` fixed the menubar, but made it so that the napari viewer would show up behind the active terminal or browser window, sometimes making it hard to notice a viewer has been opened (https://github.com/napari/napari/issues/735).  So `raise_` was readded in #721.
- #732 pointed out that `raise_` and menubar focus seem to conflict with each other... and #795 was introduced to fix the behavior for the majority of use cases.

It appears now, that OS X Catalina has the inactive menubar issue regardless of whether `raise_` is called or not.  To be clear, this problem is a new OS-version-related issue, and no commit in napari's history is free of the inactive menubar problem.  Here is the simplest example that reproduces the problem:

```python
from qtpy.QtWidgets import QMainWindow, QApplication

if __name__ == '__main__':
    app = QApplication([])
    win = QMainWindow()
    win.show()
    # win.raise_()  # doesn't really matter
    app.exec_()

# the "python" menu in the menubar will not be clickable
# until you focus away from the app and come back
```

potentially useful links (though not Catalina specific):
https://github.com/chewing/chewing-editor/issues/127
https://github.com/nsf-lrose/lrose-blaze/issues/3
https://doc.qt.io/qt-5/macos-issues.html#menu-bar
https://doc.qt.io/qt-5/macos-deployment.html

**Comment by cbrnr** [comment]
I'm pretty sure this bug is specific to Anaconda/Miniconda/conda. On my macOS 10.15.3, the minimal example works just fine when I use Python installed via Homebrew. However, if I run the same snippet with Anaconda Python, I can reproduce the issue. This seems to have something to do with Qt requiring a framework build of Python, which Anaconda's `python` command does not provide. However, you can use `pythonw` instead, which will invoke (or simulate, no idea) the framework build, and everything will work fine (including the minimal example).

**Comment by tlambert03** [comment]
thanks for chiming in!  very interesting.

**Comment by jni** [comment]
@cbrnr my hero! I totally forgot about that issue! :joy: One hopes that Anaconda eventually gets on top of this. It'll almost certainly be useful when it comes time to distributing our bundled binary.

**Comment by sofroniewn** [comment]
I'd like to bump the priority of this bug @napari/core-devs - after conversation with @pwinston @ziyangczi and @justinelarsen - one forgets how confusing it is to new comers. I've been living with this bug for months now and am just so used to it, but it really is terrible for mac users.

Maybe we can revisit the comments of @cbrnr: 

> I'm pretty sure this bug is specific to Anaconda/Miniconda/conda. On my macOS 10.15.3, the minimal example works just fine when I use Python installed via Homebrew. However, if I run the same snippet with Anaconda Python, I can reproduce the issue. This seems to have something to do with Qt requiring a framework build of Python, which Anaconda's python command does not provide. However, you can use pythonw instead, which will invoke (or simulate, no idea) the framework build, and everything will work fine (including the minimal example).

and followup with that approach.

At the same time we could look into detecting if launch of application is from ipython - in which case we still want to call `raise_` and if not then we don't want to call it anymore and the menubar focus will be fine (see comments/ links above for more detail here).

**Comment by cbrnr** [comment]
See [here](https://github.com/cbrnr/mnelab/blob/master/mnelab/__main__.py) for how we worked around this issue.

**Comment by jni** [comment]
> if launch of application is from ipython - in which case we still want to call raise_

I was bugging @tlambert03 privately about this, I actually think we should only raise from a *jupyter kernel*. From IPython, it's annoying because *usually* I am in the middle of adding a few things to the viewer, so I have to click out of napari and back to IPython.

@cbrnr 

> See here for how we worked around this issue.

🤯 That is **awesome**! Thank you!!

**Comment by sofroniewn** [comment]
This can be closed by #1554, we now use the framework python build on mac if available, where this issue does not occur
